src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -425,6 +425,11 @@ public async Task<IActionResult> Edit(string contentItemId)
 
             var model = await _contentItemDisplayManager.BuildEditorAsync(contentItem, this, false);
 
+            if(!string.IsNullOrEmpty(returnUrl))
+            {
+                TempData["returnUrl"] = returnUrl;
```

Comment:
If it's for the View only then ViewData is better. I am worried that you are using TempData but we never had to use it before to support return urls.

Reply:
I I simplified returnUrl implementation
commit: 771d500f16db2ab3fad239ef79a311f1125416d9